syft/messaging/plan.py

```diff
@@ -74,210 +65,326 @@ class State(object):
     but it allows to send this sequence of operations to remote workers and to keep a
     reference to it. This way, to compute remotely this sequence of operations on some remote
     input referenced through pointers, instead of sending multiple messages you need now to send a
     single message with the references of the plan and the pointers.
+
+    All arguments are optional.
+
+    Args:
+        name: the name of the name
+        procedure: stores and manages the plan's commands
+        state: store the plan tensors like model parameters
+        include_state: if true, implies that the plan is a function, else a class. If true, the
+            state is re-integrated in the args to be accessed within the function
+        is_built: state if the plan has already been built.
+        state_ids: ids of the state elements
+        arg_ids: ids of the last argument ids (present in the procedure commands)
+        result_ids: ids of the last result ids (present in the procedure commands)
+        readable_plan: list of commands
+        blueprint: the function to be transformed into a plan
+        state_tensors: a tuple of state elements. It can be used to populate a state
+        id: state id
+        owner: state owner
+        tags: state tags
+        description: state decription
     """
 
     def __init__(
         self,
-        id: Union[str, int] = None,
-        owner: "sy.workers.BaseWorker" = None,
-        name: str = "",
+        name: str = None,
+        procedure: Procedure = None,
+        state: State = None,
+        include_state: bool = False,
+        is_built: bool = False,
+        # Optional kwargs if commands or state are not provided
         state_ids: List[Union[str, int]] = None,
         arg_ids: List[Union[str, int]] = None,
         result_ids: List[Union[str, int]] = None,
         readable_plan: List = None,
         blueprint=None,
-        state=None,
-        include_state: bool = False,
-        is_built: bool = False,
-        verbose: bool = False,
-        *args,
-        **kwargs,
+        state_tensors=None,
+        # General kwargs
+        id: Union[str, int] = None,
+        owner: "sy.workers.BaseWorker" = None,
+        tags: List[str] = None,
+        description: str = None,
     ):
+        owner = owner or sy.local_worker
+        AbstractObject.__init__(self, id, owner, tags, description, child=None)
         ObjectStorage.__init__(self)
-        torch.nn.Module.__init__(self)
 
         # Plan instance info
-        self.id = sy.ID_PROVIDER.pop() if id is None else id
-        self.name = self.__class__.__name__ if name == "" else name
-        self._owner = sy.local_worker if owner is None else owner
-        self.verbose = verbose
-
-        # Info about the plan stored
-        self.plan = list()
-        self.readable_plan = readable_plan if readable_plan is not None else []
-        self.state = State(self)
+        self.name = name or self.__class__.__name__
```

Comment:
same as below. That being said the more I look at this the more it is growing on me

Reply:
I'm in love with this "new" syntax :D

Reply:
+1 to the `x or X()` syntax, this is pretty pythonic imo